src/melee/ft/chara/ftPurin/ftPr_Init.c

```diff
@@ -902,70 +787,27 @@ void ftPr_SpecialHi_8013CD34(HSD_GObj* gobj)
     fp->cb.x21BC_callback_Accessory4 = ftPr_Init_8013C94C;
 }
 
-#ifdef MWERKS_GEKKO
-#pragma push
-asm void ftPr_SpecialHi_8013CDD8(HSD_GObj*)
-{ // clang-format off
-    nofralloc
-/* 8013CDD8 001399B8  7C 08 02 A6 */	mflr r0
-/* 8013CDDC 001399BC  90 01 00 04 */	stw r0, 4(r1)
-/* 8013CDE0 001399C0  94 21 FF E8 */	stwu r1, -24(r1)
-/* 8013CDE4 001399C4  93 E1 00 14 */	stw r31, 20(r1)
-/* 8013CDE8 001399C8  93 C1 00 10 */	stw r30, 16(r1)
-/* 8013CDEC 001399CC  7C 7E 1B 78 */	mr r30, r3
-/* 8013CDF0 001399D0  83 E3 00 2C */	lwz r31, 44(r3)
-/* 8013CDF4 001399D4  7F E3 FB 78 */	mr r3, r31
-/* 8013CDF8 001399D8  4B F4 0A 05 */	bl ftCommon_8007D7FC
-/* 8013CDFC 001399DC  C0 22 A2 30 */	lfs f1, ftPr_Init_804D9C10(r2)
-/* 8013CE00 001399E0  C0 1F 00 2C */	lfs f0, 44(r31)
-/* 8013CE04 001399E4  FC 01 00 00 */	fcmpu cr0, f1, f0
-/* 8013CE08 001399E8  40 82 00 2C */	bne lbl_8013CE34
-/* 8013CE0C 001399EC  3C 80 0C 4C */	lis r4, 0x0C4C508A@ha
-/* 8013CE10 001399F0  C0 3F 08 94 */	lfs f1, 2196(r31)
-/* 8013CE14 001399F4  C0 42 A2 38 */	lfs f2, ftPr_Init_804D9C18(r2)
-/* 8013CE18 001399F8  7F C3 F3 78 */	mr r3, r30
-/* 8013CE1C 001399FC  C0 62 A2 34 */	lfs f3, ftPr_Init_804D9C14(r2)
-/* 8013CE20 00139A00  38 A4 50 8A */	addi r5, r4, 0x0C4C508A@l
-/* 8013CE24 00139A04  38 80 01 6D */	li r4, 365
-/* 8013CE28 00139A08  38 C0 00 00 */	li r6, 0
-/* 8013CE2C 00139A0C  4B F2 C5 81 */	bl Fighter_ChangeMotionState
-/* 8013CE30 00139A10  48 00 00 28 */	b lbl_8013CE58
-lbl_8013CE34:
-/* 8013CE34 00139A14  3C 80 0C 4C */	lis r4, 0x0C4C508A@ha
-/* 8013CE38 00139A18  C0 3F 08 94 */	lfs f1, 2196(r31)
-/* 8013CE3C 00139A1C  C0 42 A2 38 */	lfs f2, ftPr_Init_804D9C18(r2)
-/* 8013CE40 00139A20  7F C3 F3 78 */	mr r3, r30
-/* 8013CE44 00139A24  C0 62 A2 34 */	lfs f3, ftPr_Init_804D9C14(r2)
-/* 8013CE48 00139A28  38 A4 50 8A */	addi r5, r4, 0x0C4C508A@l
-/* 8013CE4C 00139A2C  38 80 01 6F */	li r4, 367
-/* 8013CE50 00139A30  38 C0 00 00 */	li r6, 0
-/* 8013CE54 00139A34  4B F2 C5 59 */	bl Fighter_ChangeMotionState
-lbl_8013CE58:
-/* 8013CE58 00139A38  3C 60 80 14 */	lis r3, ftPr_Init_8013C94C@ha
-/* 8013CE5C 00139A3C  38 03 C9 4C */	addi r0, r3, ftPr_Init_8013C94C@l
-/* 8013CE60 00139A40  90 1F 21 BC */	stw r0, 8636(r31)
-/* 8013CE64 00139A44  80 01 00 1C */	lwz r0, 28(r1)
-/* 8013CE68 00139A48  83 E1 00 14 */	lwz r31, 20(r1)
-/* 8013CE6C 00139A4C  83 C1 00 10 */	lwz r30, 16(r1)
-/* 8013CE70 00139A50  38 21 00 18 */	addi r1, r1, 24
-/* 8013CE74 00139A54  7C 08 03 A6 */	mtlr r0
-/* 8013CE78 00139A58  4E 80 00 20 */	blr
-} // clang-format on
-#pragma pop
-#endif
+void ftPr_SpecialHi_8013CDD8(Fighter_GObj* fighter_gobj)
+{
+    Fighter* fighter = fighter_gobj->user_data;
+    ftCommon_8007D7FC(fighter);
+    if (ftPr_Init_804D9C10 == fighter->facing_dir) {
+        Fighter_ChangeMotionState(fighter_gobj, 0x16D, 0x0C4C508AU,
```

Comment:
You can use the `melee-flags` tool to split out the flags here. If you don't want to bother with that, I would prefer you at least use enum values for the motion id. In this case it's `ftPr_MS_SpecialHiL`.